src/main/java/com/minecolonies/coremod/entity/ai/citizen/cook/EntityAIWorkCook.java

```diff
@@ -360,212 +142,16 @@ private AIState serveFoodToCitizen()
-
-        if (walkTo == null)
-        {
-            for (final BlockPos pos : ((BuildingCook) getOwnBuilding()).getFurnaces())
-            {
-                final TileEntity entity = world.getTileEntity(pos);
-                if (entity instanceof TileEntityFurnace && ItemStackUtils.isEmpty(((TileEntityFurnace) entity).getStackInSlot(COOK_SLOT)))
-                {
-                    walkTo = pos;
-                }
-            }
-        }
-
-        if (walkTo == null)
-        {
-            return START_WORKING;
-        }
-
-        if (walkToBlock(walkTo))
-        {
-            setDelay(2);
-            return getState();
-        }
-
-        final TileEntity entity = world.getTileEntity(walkTo);
-        if (entity instanceof TileEntityFurnace)
-        {
-            // TODO: if the furnace isn't cooking food, we should complain about that
-
-            // TODO: if the furnace is no longer able to accept our uncooked food, should we keep feeding it fuel?
-            // TODO: we should probably empty out a furnace that isn't cooking food.
-
-            InventoryUtils.transferXOfFirstSlotInItemHandlerWithIntoInItemHandler(
-                    new InvWrapper(worker.getInventoryCitizen()), ItemStackUtils.ISCOOKABLE, Constants.STACKSIZE,
-                    new InvWrapper((TileEntityFurnace) entity), COOK_SLOT);
-
-            if (ItemStackUtils.isEmpty(((TileEntityFurnace) entity).getStackInSlot(FUEL_SLOT)))
-            {
-                if (!InventoryUtils.hasItemInItemHandler(new InvWrapper(worker.getInventoryCitizen()), TileEntityFurnace::isItemFuel))
-                {
-                    walkTo = null;
-                    return COOK_GATHER_FUEL;
-                }
-
-                InventoryUtils.transferXOfFirstSlotInItemHandlerWithIntoInItemHandler(
-                        new InvWrapper(worker.getInventoryCitizen()), TileEntityFurnace::isItemFuel, Constants.STACKSIZE,
-                        new InvWrapper((TileEntityFurnace) entity), FUEL_SLOT);
-            }
-
-            walkTo = null;
-            return START_WORKING;
-        }
-        walkTo = null;
-        setDelay(STANDARD_DELAY);
-        // TODO: why COOK_COOK_FOOD instead of getState()?
-        return COOK_COOK_FOOD;
-    }
-
-    /**
-     * Check if something is waiting to be cooked in at least one oven.
-     *
-     * @return true if so.
+     * Checks if the cook has anything important to do before going to the default furnace user jobs.
+     * First calculate the building range if not cached yet.
+     * Then check for citizens around the building.
+     * If no citizen around switch to default jobs.
+     * If citizens around check if food in inventory, if not, switch to gather job.
+     * If food in inventory switch to serve job.
+     * @return the next AIState to transfer to.
      */
-    private boolean isSomethingInOven()
+    protected AIState checkForImportantJobs()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the "@Override" annotation above this method signature [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1161)